sdk/messaging/azeventhubs/checkpointstore/blob_checkpoint_store.go

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package checkpointstore
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/internal/blob"
+)
+
+// BlobCheckpointStore is a CheckpointStore implementation that uses Azure Blob storage.
+type BlobCheckpointStore struct {
+	cc *blob.ContainerClient
+}
+
+// NewOptions contains optional parameters for the New, NewFromConnectionString and NewWithSharedKey
+// functions
+type NewOptions struct {
+	azcore.ClientOptions
+}
+
+// New creates a checkpoint store that stores ownership and checkpoints in
+// Azure Blob storage. The container client will be used to store blobs.
+// NOTE: the container must exist before the checkpoint store can be used.
+func New(containerURL string, cred azcore.TokenCredential, options *NewOptions) (*BlobCheckpointStore, error) {
```

Comment:
This should be `NewBlobCheckpointStore()` or whatever name we end up landing on.

Reply:
NewBlobStore seems okay.